File: ADT/dfs_bfs/841.py
class Solution:
    def dfs(self, rooms: List[List[int]], source: int, visited: List[bool]) -> None:
        visited[source] = True
        for node in rooms[source]:
            if not visited[node]:
                self.dfs(rooms, node, visited)

    def canVisitAllRooms(self, rooms: List[List[int]]) -> bool:
        n = len(rooms)
        visited = [False] * n
        self.dfs(rooms, 0, visited)
        return all(visited)

File: ADT/dfs_bfs/841.js
/**
 * @param {number[][]} rooms
 * @return {boolean}
 */
var canVisitAllRooms = function (rooms) {
    const dfs = (source, visited) => {
        visited[source] = true
        for (const node of rooms[source]) {
            if (!visited[node]) {
                dfs(node, visited)
            }
        }
    }
    const n = rooms.length
    const visited = new Array(n).fill(false)
    dfs(0, visited)
    return visited.every(v => v)
};